### PR TITLE
Expect input of prefixes to KrakenUniq

### PR DESCRIPTION
**Pull request by Midnighter**
Updating the module will address the following bug in the taxprofiler pipeline https://github.com/nf-core/taxprofiler/issues/533 but requires that the pipeline be refactored to supply the new input.

### CODE REVIEW

File: modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf [hunk 1]
```diff
@@ -38,25 +40,26 @@ process KRAKENUNIQ_PRELOADEDKRAKENUNIQ {
...
     if (meta.single_end) {
+        assert sequences.size() == prefixes.size()
+        command_inputs = [sequences, prefixes].transpose().collect { seq, prefix -> "${seq} ${prefix}" }
+
         """
+        cat <<-END_INPUTS > ${command_inputs_file}
+        ${command_inputs.join('\n        ')}
+        END_INPUTS
+
         krakenuniq \\
```
Comment (jfy133) [inline modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf:55]:
```suggestion
        // Preload the database into memory
        krakenuniq \\
```

Comment (jfy133) [reply modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf:55]: This confused me at first

File: modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf [hunk 2]
```diff
@@ -38,25 +40,26 @@ process KRAKENUNIQ_PRELOADEDKRAKENUNIQ {
...
-        strip_suffix() {
-            local result=\$1
-            # Strip any file extensions.
-            echo "\${result%%.*}"
-        }
-
-        printf "%s\\n" ${sequences} | while read FASTQ; do \\
-            PREFIX="\$(strip_suffix "\${FASTQ}")"
-
+        while IFS=' ' read -r SEQ PREFIX; do
```
Comment (jfy133) [inline modules/nf-core/krakenuniq/preloadedkrakenuniq/main.nf:62]:
```suggestion
        // Execute the classification
        while IFS=' ' read -r SEQ PREFIX; do
```

File: modules/nf-core/krakenuniq/preloadedkrakenuniq/meta.yml [hunk 3]
```diff
@@ -25,6 +25,11 @@ input:
         type: file
         description: List of input files containing sequences. All of them must be either
           in FASTA or FASTQ format.
+    - prefixes:
+        type: string
+        description: >
+          List of sample identifiers or filename prefixes. Must correspond in order and
```
Comment (jfy133) [inline modules/nf-core/krakenuniq/preloadedkrakenuniq/meta.yml:31]: Maybe give a suggestio how to fix the order (maybe with `.sort`? Although I can imagine in some very rare cases someone might be weird and not name things such as pairs properly)

**Review by jfy133 — APPROVED** [review]

File: modules/nf-core/krakenuniq/preloadedkrakenuniq/meta.yml (hunk 3 above)
Comment (Midnighter) [reply modules/nf-core/krakenuniq/preloadedkrakenuniq/meta.yml:31]: I think that in the normal case, prefixes will be constructed from samples as they come out of a channel and thus they will be in the same order. I do not want to suggest something here that potentially makes the use more complicated than necessary and potentially leads to user errors.

I am in favor of creating a batching function and having that properly tested, but I don't know where we would provide that from?

**Pull request merged**